action.yml

```diff
@@ -31,6 +31,22 @@ inputs:
     description: 'Whether the action should wait for the end of the Gatling Enterprise simulation run before terminating. True by default. If set to false, some of this action''s outputs may be missing.'
     required: false
     default: 'true'
+  run_summary_enable:
+    description: 'When wait_for_run_end is true, will regularly display a summary of the ongoing run in the console. True by default.'
+    required: false
+    default: 'true'
+  run_summary_refresh_delay_constant:
+    description: 'Constant part of the delay before displaying a new summary of the ongoing run in the console, in seconds. 5 seconds by default (which is the minimum). This initial delay grows exponentially by the factor set in run_summary_refresh_delay_multiplier to make the output less verbose. Note that the underlying data is refreshed roughly every 5 seconds only, so the actual delay is approximate.'
+    required: false
+    default: '5'
+  run_summary_refresh_delay_multiplier:
+    description: 'A multiplier for the delay before displaying a new summary of the ongoing run in the console. 2 by default (minimum 1, which sets a constant delay). Every time a summary is displayed, the delay before the next one is multiplied by this value.'
+    required: false
+    default: '2'
+  run_summary_refresh_delay_max:
+    description: 'A maximum delay before displaying a new summary of the ongoing run in the console, in seconds. 300 seconds by default (minimum 5). The exponential increase of the delay will be capped to this value.'
+    required: false
+    default: '300'
```

Comment:
> Giving it more thoughts (sorry).

We can also start by just including the option to disable summaries, while we take a bit more time to figure out what else we want to do. That one at least won't have to break compatibility if we change our mind about the best scheduling strategy...

>  Should we display "next refresh in n seconds"?

Good idea, yes.

> Should we decrease the max default?
> Is exponential the most suited algorithm? Isn't weird to have 5s refresh, then 6s, then 7s, etc? Like, shouldn't we round to the closest 5s?

It's already rounded (up) to the nearest 5s (actually a bit more, due to how it's scheduled) because the status is still refreshed every 5 seconds (so that we don't wait an eternity to stop if the refresh delay gets very long, depending on how it's configured) and I didn't want to have two different schedules concurrently, to keep things simple.

I didn't go into all those details in the descriptions, only wrote "Note that the underlying data is refreshed roughly every 5 seconds only, so the actual delay is approximate." Note sure if it's enough.

The defaults I used give roughly 5s, 10s, 20s, 40s... until topping out at 300s. Not sure if that's really a good default... 🤔

Reply:
Fine with me then.

> Should we display "next refresh in n seconds"?
> Good idea, yes.

Want to give it a try and check if t comes up nicely or weird?

Reply:
We currently just wait 5 seconds between executions, so there is a bit of drift every time depending on how long it takes to query the API. We could schedule execution at fixed 5s intervals with `setInterval` if needed; IIRC I didn't do it because correct error handling seemed a bit trickier while executing exactly every 5 seconds didn't seem that important.

Also note that the elapsed time is based on the actual run information from the API, not on how long this process has run.

Reply:
Pull request updated, here's what I now get with a 5 minutes run and the default settings:

![Capture d’écran 2023-04-19 à 15 13 26](https://user-images.githubusercontent.com/6560742/233085738-06035bea-d1ed-4ca2-8211-b45a17fbba06.png)